### PR TITLE
feat: DAH-2451 cache service gets translations on listing translations cache miss

### PR DESCRIPTION
**Pull request by cade-exygy**
## Description

When we hit the listing detail service, we return the listing translations alongside the listing. If the listing translations do not exist, we need to call the translation service to grab them and cache them.

## Jira ticket
DAH-2451
relies on #2244 being merged

### Version Control

- [x] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [x] [if the set of changes cannot be small, reviewers have been forewarned](https://google.github.io/eng-practices/review/developer/small-cls.html#cant)
- [x] code meets test coverage thresholds
- [x] code is properly formatted
- [x] code is linted
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] automated tests pass
- [x] if the code changes the UI, it matches the UI design exactly

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [x] If time sensitive, notify engineers in Slack

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions have already been performed at least once
- [x] instructions can be followed by PA testers
- [x] instructions specify if it can only be followed by an engineer

On localhost:
run memcached
Go to a listing
Check logs for `CacheService.processTranslations` being called
Inspect in Network Tab the Listing response and ensure the translation object is populated.

If you refresh, you should not see this call happen again because the `cache.fetch()` call should hit successfully.

### CODE REVIEW

**Review by tallulahkay — APPROVED** [review]
This works well!! I love how small this ticket is - made it really easy to review 🤩 

I worry we are clogging up the logs because the translation object logs are so big.. maybe we can shorten to just success or failure messages, or just log the keys or somethin

**Comment by tallulahkay** [comment]
What if this went in first? Would we hit the google translation api too much?

**Comment by tallulahkay** [comment]
@cade-exygy I found an issue with this!! if the cache is not running, it doesn't return the translations

**Comment by cade-exygy** [comment]
> @cade-exygy I found an issue with this!! if the cache is not running, it doesn't return the translations I think it needs to return translations whether or not it successfully caches them in `cache_listing_translations`

Nice catch! I put that fix in #2244 to keep these changes focused on the listing service

**Comment by tallulahkay** [comment]
This doesn't _quite_ work because if there are translations for a listing, but incomplete ones (like for only one field, for ex) then it doesn't get the translations. We need to do some sort of check to make sure the cache has all the translations we need

**Review by tallulahkay — APPROVED** [review]
lgtm after the test is updated!

As soon as we merge this, we will start hitting the google cloud translate api - is full prepared for that?

File: spec/services/cache_service_spec.rb
Comment (jimlin-sfgov) [inline spec/services/cache_service_spec.rb:138]: nit: I'm assuming this was for debugging, we should remove them from tests.

**Review by jimlin-sfgov — APPROVED** [review]
LG

one nit suggestion

**Pull request merged**
